### ISSUE TITLE
MySQL DB for PIWIK reached max connection limit and PIWIK stops working.

### ISSUE THREAD

**Issue by samratng07** [piwik/piwik#10222]
Hi,

is there any max db connection limit in PIWIK configuration,can I set the max number of connections in my PIWIK configurations,as MySQL DB for PIWIK reached max connection limit of 600 and PIWIK stops working bringing the entire site down.

If I can prevent this thing from happening by making changes to PIWIK configurations it would be a great help.

**Comment by tsteur** [comment]
Does this happen because of too many tracking requests? Have a look at https://github.com/piwik/plugin-QueuedTracking this would help in such a case and reduce database connections drastically

**Comment by samratng07** [comment]
yes thank you,we will have a look into this try to implement it and then confirm with you,till then please keep the issue open

**Comment by mattab** [comment]
Maybe we could create a new system check that does basic Mysql settings configuration checks, and recommends to increase the max connection when applicable...? 

@samratng07 has QueuedTracking solved the issue for you?

**Comment by gauravtyagi** [comment]
Implementing Queue Tracking slows down the insertion rate, can we go with the DB pooling or connection reuse.
Are we opening and closing the connection after each DB transaction.
Approx how many connections are required to serve ~1k requests/sec
